game/lib/leaderboard/leaderboard_widget.dart

```diff
@@ -24,18 +29,24 @@ class _LeaderboardWidgetState extends State<LeaderboardWidget> {
         child: Container(
           child: Padding(
             padding: const EdgeInsets.only(left: 8.0, right: 8.0),
-            child: ListView(
-              shrinkWrap: true,
-              scrollDirection: Axis.vertical,
-              children: [
-                leaderBoardUserCell(context, "User", "1", "6", "23"),
-                leaderBoardCell(context, "User", "1", "23", true),
-                leaderBoardCell(context, "User 2", "2", "20", false),
-                leaderBoardCell(context, "User 3", "3", "19", false),
-                leaderBoardCell(context, "User 4", "4", "19", false),
-                leaderBoardCell(context, "User 5", "5", "19", false),
-                leaderBoardCell(context, "User 6", "6", "19", false),
-              ],
+            child: Consumer2<GroupModel, EventModel>(
+              builder: (context, myGroupModel, myEventModel, child) {
+                if (myGroupModel.curEventId == null) return ListView();
+                final List<UpdateLeaderDataUserDto> list = myEventModel
+                    .getTopPlayersForEvent(myGroupModel.curEventId!, 6);
+                return ListView(
+                  shrinkWrap: true,
+                  scrollDirection: Axis.vertical,
+                  children: [
+                    for (int i = 0; i < min(6, list.length); i++)
+                      leaderBoardUserCell(context, list.elementAt(i).username,
```

Comment:
This cell actually needs to occur only once at the very top, since this is actually the current user.